> I'm not sure I totally understand why GCC is flagging this or why taking the value by reference would help.

To be honest, I don't quite understand why GCC is complaining.

> I would love to see the complete error message (assuming there is more than what is in the bugzilla), because this is quite an interesting warning.

Here is the full error message:
```
[build] In file included from /home/vitaly/WebKit/Source/WebCore/css/CSSPrimitiveValue.h:26,
[build]                  from /home/vitaly/WebKit/Source/WebCore/css/parser/CSSParserToken.h:32,
[build]                  from /home/vitaly/WebKit/Source/WebCore/css/parser/CSSPropertyParserConsumer+TimeDefinitions.h:27,
[build]                  from /home/vitaly/WebKit/Source/WebCore/css/parser/CSSPropertyParserConsumer+Time.cpp:27,
[build]                  from /home/vitaly/WebKit/WebKitBuild/GTK/Release/WebCore/DerivedSources/unified-sources/UnifiedSource-2f84417a-24.cpp:1:
[build] In member function 'void WebCore::CSSValue::deref() const',
[build]     inlined from 'static void WTF::DefaultRefDerefTraits< <template-parameter-1-1> >::derefIfNotNull(T*) [with T = WebCore::CSSCalcValue]' at /home/vitaly/WebKit/WebKitBuild/GTK/Release/WTF/Headers/wtf/Ref.h:62:23,
[build]     inlined from 'WTF::Ref<T, <template-parameter-1-2>, <template-parameter-1-3> >::~Ref() [with T = WebCore::CSSCalcValue; _PtrTraits = WTF::RawPtrTraits<WebCore::CSSCalcValue>; RefDerefTraits = WTF::DefaultRefDerefTraits<WebCore::CSSCalcValue>]' at /home/vitaly/WebKit/WebKitBuild/GTK/Release/WTF/Headers/wtf/Ref.h:82:43,
[build]     inlined from 'WebCore::UnevaluatedCalc<WebCore::TimeRaw>::~UnevaluatedCalc()' at /home/vitaly/WebKit/Source/WebCore/css/parser/CSSPropertyParserConsumer+UnevaluatedCalc.h:46:29,
[build]     inlined from 'constexpr _Res std::__invoke_impl(__invoke_other, _Fn&&, _Args&& ...) [with _Res = WTF::RefPtr<WebCore::CSSPrimitiveValue>; _Fn = WTF::Visitor<WebCore::CSSPropertyParserHelpers::MetaResolver<WTF::RefPtr<WebCore::CSSPrimitiveValue>, WebCore::CSSPropertyParserHelpers::CSSPrimitiveValueResolverBase, WebCore::TimeRaw>::resolve(const WebCore::CSSPropertyParserHelpers::MetaConsumeResult<WebCore::TimeRaw>::type&, const WebCore::CSSCalcSymbolTable&, WebCore::CSSPropertyParserHelpers::CSSPropertyParserOptions)::<lambda(auto:64)> >; _Args = {const WebCore::UnevaluatedCalc<WebCore::TimeRaw>&}]' at /usr/include/c++/12/bits/invoke.h:61:67,
[build]     inlined from 'constexpr typename std::__invoke_result<_Functor, _ArgTypes>::type std::__invoke(_Callable&&, _Args&& ...) [with _Callable = WTF::Visitor<WebCore::CSSPropertyParserHelpers::MetaResolver<WTF::RefPtr<WebCore::CSSPrimitiveValue>, WebCore::CSSPropertyParserHelpers::CSSPrimitiveValueResolverBase, WebCore::TimeRaw>::resolve(const WebCore::CSSPropertyParserHelpers::MetaConsumeResult<WebCore::TimeRaw>::type&, const WebCore::CSSCalcSymbolTable&, WebCore::CSSPropertyParserHelpers::CSSPropertyParserOptions)::<lambda(auto:64)> >; _Args = {const WebCore::UnevaluatedCalc<WebCore::TimeRaw>&}]' at /usr/include/c++/12/bits/invoke.h:97:36,
[build]     inlined from 'static constexpr decltype(auto) std::__detail::__variant::__gen_vtable_impl<std::__detail::__variant::_Multi_array<_Result_type (*)(_Visitor, _Variants ...)>, std::integer_sequence<long unsigned int, __indices ...> >::__visit_invoke(_Visitor&&, _Variants ...) [with _Result_type = std::__detail::__variant::__deduce_visit_result<WTF::RefPtr<WebCore::CSSPrimitiveValue> >; _Visitor = WTF::Visitor<WebCore::CSSPropertyParserHelpers::MetaResolver<WTF::RefPtr<WebCore::CSSPrimitiveValue>, WebCore::CSSPropertyParserHelpers::CSSPrimitiveValueResolverBase, WebCore::TimeRaw>::resolve(const WebCore::CSSPropertyParserHelpers::MetaConsumeResult<WebCore::TimeRaw>::type&, const WebCore::CSSCalcSymbolTable&, WebCore::CSSPropertyParserHelpers::CSSPropertyParserOptions)::<lambda(auto:64)> >&&; _Variants = {const std::variant<WebCore::TimeRaw, WebCore::UnevaluatedCalc<WebCore::TimeRaw> >&}; long unsigned int ...__indices = {1}]' at /usr/include/c++/12/variant:1033:38,
[build]     inlined from 'constexpr decltype(auto) std::__do_visit(_Visitor&&, _Variants&& ...) [with _Result_type = __detail::__variant::__deduce_visit_result<WTF::RefPtr<WebCore::CSSPrimitiveValue> >; _Visitor = WTF::Visitor<WebCore::CSSPropertyParserHelpers::MetaResolver<WTF::RefPtr<WebCore::CSSPrimitiveValue>, WebCore::CSSPropertyParserHelpers::CSSPrimitiveValueResolverBase, WebCore::TimeRaw>::resolve(const WebCore::CSSPropertyParserHelpers::MetaConsumeResult<WebCore::TimeRaw>::type&, const WebCore::CSSCalcSymbolTable&, WebCore::CSSPropertyParserHelpers::CSSPropertyParserOptions)::<lambda(auto:64)> >; _Variants = {const variant<WebCore::TimeRaw, WebCore::UnevaluatedCalc<WebCore::TimeRaw> >&}]' at /usr/include/c++/12/variant:1791:164,
[build]     inlined from 'constexpr std::__detail::__variant::__visit_result_t<_Visitor, _Variants ...> std::visit(_Visitor&&, _Variants&& ...) [with _Visitor = WTF::Visitor<WebCore::CSSPropertyParserHelpers::MetaResolver<WTF::RefPtr<WebCore::CSSPrimitiveValue>, WebCore::CSSPropertyParserHelpers::CSSPrimitiveValueResolverBase, WebCore::TimeRaw>::resolve(const WebCore::CSSPropertyParserHelpers::MetaConsumeResult<WebCore::TimeRaw>::type&, const WebCore::CSSCalcSymbolTable&, WebCore::CSSPropertyParserHelpers::CSSPropertyParserOptions)::<lambda(auto:64)> >; _Variants = {const variant<WebCore::TimeRaw, WebCore::UnevaluatedCalc<WebCore::TimeRaw> >&}]' at /usr/include/c++/12/variant:1855:39,
[build]     inlined from 'decltype (std::visit(WTF::makeVisitor((forward<F>)(WTF::switchOn::f)...), forward<V>(v))) WTF::switchOn(V&&, F&& ...) [with V = const std::variant<WebCore::TimeRaw, WebCore::UnevaluatedCalc<WebCore::TimeRaw> >&; F = {WebCore::CSSPropertyParserHelpers::MetaResolver<RefPtr<WebCore::CSSPrimitiveValue>, WebCore::CSSPropertyParserHelpers::CSSPrimitiveValueResolverBase, WebCore::TimeRaw>::resolve(const WebCore::CSSPropertyParserHelpers::MetaConsumeResult<WebCore::TimeRaw>::type&, const WebCore::CSSCalcSymbolTable&, WebCore::CSSPropertyParserHelpers::CSSPropertyParserOptions)::<lambda(auto:64)>}]' at /home/vitaly/WebKit/WebKitBuild/GTK/Release/WTF/Headers/wtf/StdLibExtras.h:448:77,
[build]     inlined from 'static WebCore::CSSPropertyParserHelpers::MetaResolver<R, Base, Ts>::ResultType WebCore::CSSPropertyParserHelpers::MetaResolver<R, Base, Ts>::resolve(const typename WebCore::CSSPropertyParserHelpers::MetaConsumeResult<Ts ...>::type&, const WebCore::CSSCalcSymbolTable&, WebCore::CSSPropertyParserHelpers::CSSPropertyParserOptions) [with R = WTF::RefPtr<WebCore::CSSPrimitiveValue>; Base = WebCore::CSSPropertyParserHelpers::CSSPrimitiveValueResolverBase; Ts = {WebCore::TimeRaw}]' at /home/vitaly/WebKit/Source/WebCore/css/parser/CSSPropertyParserConsumer+MetaResolver.h:52:10,
[build]     inlined from 'static WebCore::CSSPropertyParserHelpers::MetaResolver<R, Base, Ts>::ResultType WebCore::CSSPropertyParserHelpers::MetaResolver<R, Base, Ts>::consumeAndResolve(WebCore::CSSParserTokenRange&, WebCore::CSSCalcSymbolsAllowed, const WebCore::CSSCalcSymbolTable&, WebCore::CSSPropertyParserHelpers::CSSPropertyParserOptions) [with R = WTF::RefPtr<WebCore::CSSPrimitiveValue>; Base = WebCore::CSSPropertyParserHelpers::CSSPrimitiveValueResolverBase; Ts = {WebCore::TimeRaw}]' at /home/vitaly/WebKit/Source/WebCore/css/parser/CSSPropertyParserConsumer+MetaResolver.h:60:89,
[build]     inlined from 'WTF::RefPtr<WebCore::CSSPrimitiveValue> WebCore::CSSPropertyParserHelpers::consumeTime(WebCore::CSSParserTokenRange&, WebCore::CSSParserMode, WebCore::ValueRange, UnitlessQuirk)' at /home/vitaly/WebKit/Source/WebCore/css/parser/CSSPropertyParserConsumer+Time.cpp:113:90:
[build] /home/vitaly/WebKit/Source/WebCore/css/CSSValue.h:341:16: error: pointer may be used after 'static void WebCore::CSSValue::operator delete(WebCore::CSSValue*, std::destroying_delete_t)' [-Werror=use-after-free]
[build] In member function 'void WebCore::CSSValue::deref() const',
[build]     inlined from 'static void WTF::DefaultRefDerefTraits< <template-parameter-1-1> >::derefIfNotNull(T*) [with T = WebCore::CSSCalcValue]' at /home/vitaly/WebKit/WebKitBuild/GTK/Release/WTF/Headers/wtf/Ref.h:62:23,
[build]     inlined from 'WTF::Ref<T, <template-parameter-1-2>, <template-parameter-1-3> >::~Ref() [with T = WebCore::CSSCalcValue; _PtrTraits = WTF::RawPtrTraits<WebCore::CSSCalcValue>; RefDerefTraits = WTF::DefaultRefDerefTraits<WebCore::CSSCalcValue>]' at /home/vitaly/WebKit/WebKitBuild/GTK/Release/WTF/Headers/wtf/Ref.h:82:43,
[build]     inlined from 'WebCore::UnevaluatedCalc<WebCore::TimeRaw>::~UnevaluatedCalc()' at /home/vitaly/WebKit/Source/WebCore/css/parser/CSSPropertyParserConsumer+UnevaluatedCalc.h:46:29,
[build]     inlined from 'WebCore::CSSPropertyParserHelpers::MetaResolver<WTF::RefPtr<WebCore::CSSPrimitiveValue>, WebCore::CSSPropertyParserHelpers::CSSPrimitiveValueResolverBase, WebCore::TimeRaw>::resolve(const WebCore::CSSPropertyParserHelpers::MetaConsumeResult<WebCore::TimeRaw>::type&, const WebCore::CSSCalcSymbolTable&, WebCore::CSSPropertyParserHelpers::CSSPropertyParserOptions)::<lambda(auto:64)> [with auto:64 = WebCore::UnevaluatedCalc<WebCore::TimeRaw>]' at /home/vitaly/WebKit/Source/WebCore/css/parser/CSSPropertyParserConsumer+MetaResolver.h:51:61,
[build]     inlined from 'constexpr _Res std::__invoke_impl(__invoke_other, _Fn&&, _Args&& ...) [with _Res = WTF::RefPtr<WebCore::CSSPrimitiveValue>; _Fn = WTF::Visitor<WebCore::CSSPropertyParserHelpers::MetaResolver<WTF::RefPtr<WebCore::CSSPrimitiveValue>, WebCore::CSSPropertyParserHelpers::CSSPrimitiveValueResolverBase, WebCore::TimeRaw>::resolve(const WebCore::CSSPropertyParserHelpers::MetaConsumeResult<WebCore::TimeRaw>::type&, const WebCore::CSSCalcSymbolTable&, WebCore::CSSPropertyParserHelpers::CSSPropertyParserOptions)::<lambda(auto:64)> >; _Args = {const WebCore::UnevaluatedCalc<WebCore::TimeRaw>&}]' at /usr/include/c++/12/bits/invoke.h:61:67,
[build]     inlined from 'constexpr typename std::__invoke_result<_Functor, _ArgTypes>::type std::__invoke(_Callable&&, _Args&& ...) [with _Callable = WTF::Visitor<WebCore::CSSPropertyParserHelpers::MetaResolver<WTF::RefPtr<WebCore::CSSPrimitiveValue>, WebCore::CSSPropertyParserHelpers::CSSPrimitiveValueResolverBase, WebCore::TimeRaw>::resolve(const WebCore::CSSPropertyParserHelpers::MetaConsumeResult<WebCore::TimeRaw>::type&, const WebCore::CSSCalcSymbolTable&, WebCore::CSSPropertyParserHelpers::CSSPropertyParserOptions)::<lambda(auto:64)> >; _Args = {const WebCore::UnevaluatedCalc<WebCore::TimeRaw>&}]' at /usr/include/c++/12/bits/invoke.h:97:36,
[build]     inlined from 'static constexpr decltype(auto) std::__detail::__variant::__gen_vtable_impl<std::__detail::__variant::_Multi_array<_Result_type (*)(_Visitor, _Variants ...)>, std::integer_sequence<long unsigned int, __indices ...> >::__visit_invoke(_Visitor&&, _Variants ...) [with _Result_type = std::__detail::__variant::__deduce_visit_result<WTF::RefPtr<WebCore::CSSPrimitiveValue> >; _Visitor = WTF::Visitor<WebCore::CSSPropertyParserHelpers::MetaResolver<WTF::RefPtr<WebCore::CSSPrimitiveValue>, WebCore::CSSPropertyParserHelpers::CSSPrimitiveValueResolverBase, WebCore::TimeRaw>::resolve(const WebCore::CSSPropertyParserHelpers::MetaConsumeResult<WebCore::TimeRaw>::type&, const WebCore::CSSCalcSymbolTable&, WebCore::CSSPropertyParserHelpers::CSSPropertyParserOptions)::<lambda(auto:64)> >&&; _Variants = {const std::variant<WebCore::TimeRaw, WebCore::UnevaluatedCalc<WebCore::TimeRaw> >&}; long unsigned int ...__indices = {1}]' at /usr/include/c++/12/variant:1033:38,
[build]     inlined from 'constexpr decltype(auto) std::__do_visit(_Visitor&&, _Variants&& ...) [with _Result_type = __detail::__variant::__deduce_visit_result<WTF::RefPtr<WebCore::CSSPrimitiveValue> >; _Visitor = WTF::Visitor<WebCore::CSSPropertyParserHelpers::MetaResolver<WTF::RefPtr<WebCore::CSSPrimitiveValue>, WebCore::CSSPropertyParserHelpers::CSSPrimitiveValueResolverBase, WebCore::TimeRaw>::resolve(const WebCore::CSSPropertyParserHelpers::MetaConsumeResult<WebCore::TimeRaw>::type&, const WebCore::CSSCalcSymbolTable&, WebCore::CSSPropertyParserHelpers::CSSPropertyParserOptions)::<lambda(auto:64)> >; _Variants = {const variant<WebCore::TimeRaw, WebCore::UnevaluatedCalc<WebCore::TimeRaw> >&}]' at /usr/include/c++/12/variant:1791:164,
[build]     inlined from 'constexpr std::__detail::__variant::__visit_result_t<_Visitor, _Variants ...> std::visit(_Visitor&&, _Variants&& ...) [with _Visitor = WTF::Visitor<WebCore::CSSPropertyParserHelpers::MetaResolver<WTF::RefPtr<WebCore::CSSPrimitiveValue>, WebCore::CSSPropertyParserHelpers::CSSPrimitiveValueResolverBase, WebCore::TimeRaw>::resolve(const WebCore::CSSPropertyParserHelpers::MetaConsumeResult<WebCore::TimeRaw>::type&, const WebCore::CSSCalcSymbolTable&, WebCore::CSSPropertyParserHelpers::CSSPropertyParserOptions)::<lambda(auto:64)> >; _Variants = {const variant<WebCore::TimeRaw, WebCore::UnevaluatedCalc<WebCore::TimeRaw> >&}]' at /usr/include/c++/12/variant:1855:39,
[build]     inlined from 'decltype (std::visit(WTF::makeVisitor((forward<F>)(WTF::switchOn::f)...), forward<V>(v))) WTF::switchOn(V&&, F&& ...) [with V = const std::variant<WebCore::TimeRaw, WebCore::UnevaluatedCalc<WebCore::TimeRaw> >&; F = {WebCore::CSSPropertyParserHelpers::MetaResolver<RefPtr<WebCore::CSSPrimitiveValue>, WebCore::CSSPropertyParserHelpers::CSSPrimitiveValueResolverBase, WebCore::TimeRaw>::resolve(const WebCore::CSSPropertyParserHelpers::MetaConsumeResult<WebCore::TimeRaw>::type&, const WebCore::CSSCalcSymbolTable&, WebCore::CSSPropertyParserHelpers::CSSPropertyParserOptions)::<lambda(auto:64)>}]' at /home/vitaly/WebKit/WebKitBuild/GTK/Release/WTF/Headers/wtf/StdLibExtras.h:448:77,
[build]     inlined from 'static WebCore::CSSPropertyParserHelpers::MetaResolver<R, Base, Ts>::ResultType WebCore::CSSPropertyParserHelpers::MetaResolver<R, Base, Ts>::resolve(const typename WebCore::CSSPropertyParserHelpers::MetaConsumeResult<Ts ...>::type&, const WebCore::CSSCalcSymbolTable&, WebCore::CSSPropertyParserHelpers::CSSPropertyParserOptions) [with R = WTF::RefPtr<WebCore::CSSPrimitiveValue>; Base = WebCore::CSSPropertyParserHelpers::CSSPrimitiveValueResolverBase; Ts = {WebCore::TimeRaw}]' at /home/vitaly/WebKit/Source/WebCore/css/parser/CSSPropertyParserConsumer+MetaResolver.h:52:10,
[build]     inlined from 'static WebCore::CSSPropertyParserHelpers::MetaResolver<R, Base, Ts>::ResultType WebCore::CSSPropertyParserHelpers::MetaResolver<R, Base, Ts>::consumeAndResolve(WebCore::CSSParserTokenRange&, WebCore::CSSCalcSymbolsAllowed, const WebCore::CSSCalcSymbolTable&, WebCore::CSSPropertyParserHelpers::CSSPropertyParserOptions) [with R = WTF::RefPtr<WebCore::CSSPrimitiveValue>; Base = WebCore::CSSPropertyParserHelpers::CSSPrimitiveValueResolverBase; Ts = {WebCore::TimeRaw}]' at /home/vitaly/WebKit/Source/WebCore/css/parser/CSSPropertyParserConsumer+MetaResolver.h:60:89,
[build]     inlined from 'WTF::RefPtr<WebCore::CSSPrimitiveValue> WebCore::CSSPropertyParserHelpers::consumeTime(WebCore::CSSParserTokenRange&, WebCore::CSSParserMode, WebCore::ValueRange, UnitlessQuirk)' at /home/vitaly/WebKit/Source/WebCore/css/parser/CSSPropertyParserConsumer+Time.cpp:113:90:
[build] /home/vitaly/WebKit/Source/WebCore/css/CSSValue.h:341:16: note: call to 'static void WebCore::CSSValue::operator delete(WebCore::CSSValue*, std::destroying_delete_t)' here
```

--- a/Source/WebCore/css/parser/CSSPropertyParserConsumer+MetaResolver.h
+++ b/Source/WebCore/css/parser/CSSPropertyParserConsumer+MetaResolver.h
@@ -47,7 +47,7 @@ struct MetaResolver : Base {
 
     static ResultType resolve(const typename MetaConsumeResult<Ts...>::type& consumeResult, const CSSCalcSymbolTable& symbolTable, CSSPropertyParserOptions options)
     {
-        return WTF::switchOn(consumeResult, [&](auto value) -> ResultType {
+        return WTF::switchOn(consumeResult, [&](auto& value) -> ResultType {
             return Base::resolve(value, symbolTable, options);
         });
     }

